### DISCUSSION TITLE
QEMU CPU name in blinky-arm tutorial

### DISCUSSION THREAD

**Discussion by solomonyeUOA** [eclipse-embed-cdt/eclipse-plugins#594] [Debug Plug-ins Q&A] [answered]
I follow up https://eclipse-embed-cdt.github.io/tutorials/blinky-arm/  I have chosen device STM32F407VG for the project. When I create debug configuration, I need fill in QEMU machine name and QEMU CPU name, which is different from the tutorial. I have tried to put virt-7.2 in machine name. But each time when I click debug, there is below error. qemu-system-arm.exe: unable to find CPU model 'STM32F407VG'  How could I run this tutorial? What is the CPU model I should I fill in? which version of Eclipse I should use? I have changed the device to STM32F407VGTx, it does not work. Should I should earlier version of CMSIS packs for STM32F4XX_DFP? Currently I install 2.17.1 version. Thanks for help

**Comment by TommyMurphyTM1234** [comment]
You don't seem to be using the xPack QEMU which is named `qemu-system-gnuarmeclipse` according to this:

* https://xpack.github.io/dev-tools/qemu-arm/

According to the release notes it supports STM32F407VG.

* https://xpack.github.io/dev-tools/qemu-arm/releases/

**Comment by solomonyeUOA** [comment]
Dear Tommy, thanks for your reply above. I try  xpm install @xpack-dev-tools/qemu-arm@latest --verbose. But it is said Processing @xpack-dev-tools/qemu-arm@7.2.5-1.1...warning: package @xpack-dev-tools/qemu-arm@7.2.5-1.1 already installed; use --force to overwrite. So it is already installed. I have found STM32F407VG and assign it to the blinky project as device. But what should I put in QEMU CPU name and QEMU machine name? The tutorial does not have these 2 items. When debug, qemu-system-arm.exe: unable to find CPU model 'STM32F407VG'. Please help.

**Reply by TommyMurphyTM1234** [reply]
Sorry - I don't know. Maybe @ilg-ul can comment?

**Comment by solomonyeUOA** [comment]
In the tutorial, which version of eclipse and other software are working? Then I can download those version to try? Maybe I use the latest Eclipse version, for example, Eclipse IDE for Embedded C/C++ Developers (includes Incubating components) Version: 2023-09 (4.29.0) and it does not work?  The tutorial version does not have QEMU CPU and machine name to fill in, for example. Thanks

**Comment by ilg-ul** [comment, marked as answer]
How about reading the existing documentation?

- https://xpack.github.io/blog/2023/09/03/qemu-arm-v7-2-5-1-released/

**Comment by solomonyeUOA** [comment]
Thanks for @ilg-ul @TommyMurphyTM1234 With both of your help and hint, it works now. I have changed the exe to qemu-system-gnuarmeclipse.exe instead of qemu-system-arm.exe. CPU name is cortex-m4 and machine name is STM32F4-Discovery. It works with simulation board. It is fantastic and both of you are amazing. Could I simulate DE2-115 board here as well? Both of them use Eclipse. Or anywhere could help me to solve the simulation of DE2-115 board?

**Comment by TommyMurphyTM1234** [comment]
> I have changed the exe to qemu-system-gnuarmeclipse.exe instead of qemu-system-arm.exe. 

That is what I suggested originally:

* https://github.com/eclipse-embed-cdt/eclipse-plugins/discussions/594#discussioncomment-7357527

> Could I simulate DE2-115 board here as well? Both of them use Eclipse. 

I don't know what you mean by "both of them use Eclipse"? Both of what? The DE2-115 is an FPGA board and not a specific Arm MCU board so I'm not really sure what you would want QEMU to emulate? Unless you mean a specific FPGA design implementing a CPU/MCU based SoC? That might need customisation of the QEMU configuration or even extension of the QEMU implementation. That's way outside the scope of this forum.

**Comment by ilg-ul** [comment]
For the list of QEMU supported machines, see the previous link. Adding a new machine/board is not a trivial task.

**Comment by solomonyeUOA** [comment]
Great thanks for the help above. We do not need to worry DE2-115 board here since it is not on the machine list.  Really appreciate both of your help. I first try the forum or discussion and it really helps.
